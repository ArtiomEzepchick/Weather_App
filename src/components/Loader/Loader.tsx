import React from "react"

import "./index.scss"

const Loader: React.FC = () => {
    return (
        <div id="loader"></div>
    )
}

export default Loader