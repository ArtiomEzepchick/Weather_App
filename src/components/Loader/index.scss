@import '../../variables.scss';

#loader {
    position: fixed;
    top: 50%;
    left: 50%;
    border: 4px solid $color-white;
    border-top: 4px solid $color-dark-blue;
    border-radius: 50%;
    width: 3rem;
    height: 3rem;
    margin-top: -2.6rem;
    margin-left: -.5rem;
    animation: rotate 2s linear infinite;
    z-index: 500;

    &.margin-added {
        margin-left: -1.2rem;
        margin-top: -2rem;
    }

    @keyframes rotate {
        0% {
            transform: rotate(0deg);
        }
    
        100% {
            transform: rotate(360deg);
        }
    }
    
    @media (max-width: $window-m) {
        margin-left: 1rem;
    }

    @media (max-width: $window-s) {
        margin-left: 0;
    }
}