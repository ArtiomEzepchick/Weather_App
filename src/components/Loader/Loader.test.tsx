import React from "react"
import { render, screen } from "@testing-library/react"
import '@testing-library/jest-dom'

import Loader from "./Loader"

describe('Loader', () => {
    it("renders with 'margin-added' class when weatherDataLength equals 0", () => {
        render(<Loader weatherDataLength={0}/>)

        const loader = screen.getByTestId('loader')

        expect(loader).toBeInTheDocument()
        expect(loader).toHaveClass('margin-added')
        expect(loader).toMatchSnapshot()
    })

    it("renders without 'margin-added' class when weatherDataLength is more than 0", () => {
        render(<Loader weatherDataLength={1}/>)

        const loader = screen.getByTestId('loader')
        
        expect(loader).toBeInTheDocument()
        expect(loader).not.toHaveClass('margin-added')
        expect(loader).toMatchSnapshot()
    })
})