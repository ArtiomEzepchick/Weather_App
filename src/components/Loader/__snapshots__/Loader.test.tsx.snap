// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders with 'margin-added' class when weatherDataLength equals 0 1`] = `
<div
  class="margin-added"
  data-testid="loader"
  id="loader"
/>
`;

exports[`Loader renders without 'margin-added' class when weatherDataLength is more than 0 1`] = `
<div
  class=""
  data-testid="loader"
  id="loader"
/>
`;
